### PR TITLE
Add nrf_cc310_platform library

### PR DESCRIPTION
**Pull request by wackazong**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -4,12 +4,12 @@ version = "2.7.1"
...
 	"Jonathan 'theJPster' Pallant <github@thejpster.org.uk>",
 	"42 Technology Ltd <jonathan.pallant@42technology.com>",
-	"Dion Dokter <diondokter@gmail.com>"
+	"Dion Dokter <diondokter@gmail.com>",
 ]
 edition = "2021"
 description = "Rust bindings to the Nordic nRF91* Socket Library."
 readme = "README.md"
-license-file = "LICENCE.md"
+license-file = "LICENSE.md"
```
Comment (diondokter) [inline Cargo.toml]: Why the change? It was correct

File: README.md [hunk 2]
```diff
@@ -58,11 +58,11 @@ You might also prefer the async [higher-level wrapper](https://crates.io/crates/
...
 ```
 
-## Licence
+## License
 
 Any of the code outside the `./third_party` folder is under the [Blue Oak
-Licence](./LICENCE.md). Any code inside the `./third_party` folder (include
-the Nordic nrfxlib) has its own LICENCE file.
+License](./LICENSE.md). Any code inside the `./third_party` folder (include
+the Nordic nrfxlib) has its own LICENSE file.
```
Comment (diondokter) [inline README.md]: This wasn't a spelling mistake. This repo was started by a Brit and 'licence' is the way you write it in british. Please revert

File: Cargo.toml [hunk 3]
```diff
@@ -37,11 +37,11 @@ include = [
 
 [build-dependencies]
 regex = "1"
-bindgen = "0.66.1"
+bindgen = "0.70"
 llvm-tools = { version = "0.1.1", optional = true }
 
 [features]
-default = ["llvm-objcopy"]
+default = ["llvm-objcopy", "nrf9160"]
```
Comment (diondokter) [inline Cargo.toml]: While nice for nrf9160 users, this is a slight annoyance for users of other chips. It was a deliberate decision to have it as it was

File: Cargo.toml (hunk 1 above)
Comment (wackazong) [reply Cargo.toml]: That was just the spelling, this does not necessarily need to be in this pull request.

File: README.md (hunk 2 above)
Comment (wackazong) [reply README.md]: Ok

File: Cargo.toml (hunk 3 above)
Comment (wackazong) [reply Cargo.toml]: Oversight on my part.

**Comment by wackazong** [comment]
Modifications as requested.

**Review by diondokter — APPROVED** [review]
